### PR TITLE
add ceno pcs trait

### PR DESCRIPTION
**Pull request by chaosma**
- [x] add pcs trait from ceno
- [x] Adapt whir pcs code according to pcs trait

### CODE REVIEW

**Comment by yczhangsjtu** [comment]
I suggest keeping the modification as separated as possible from the existing code, because the upstream is still under active development, and we may want to periodically merge in their optimizations and bug fixes. Modifying the existing files directly may introduce a lot of changes due to cargo fmt, and more and more (unnecessary) merge conflicts.

**Comment by yczhangsjtu** [comment]
I picked the most important changes from this PR to a new branch `cyte/ceno-binding`. Would you mind rebasing your new changes to this branch?

**Comment by chaosma** [comment]
> I picked the most important changes from this PR to a new branch `cyte/ceno-binding`. Would you mind rebasing your new changes to this branch?

I have changed the pr according to ceno-binding module directly

File: src/whir/mod.rs [hunk 1]
```diff
@@ -8,7 +8,7 @@ pub mod iopattern;
 pub mod parameters;
 pub mod prover;
 pub mod verifier;
-mod fs_utils;
+pub mod fs_utils;
```
Comment (yczhangsjtu) [inline src/whir/mod.rs:11]: Why making this public? Are we using it in our own module?

Comment (chaosma) [reply src/whir/mod.rs:11]: I copied it from the latest master branch, they should be same 😂

https://github.com/scroll-tech/whir/blob/main/src/whir/mod.rs#L11

Comment (yczhangsjtu) [reply src/whir/mod.rs:11]: A bug in GitHub?

**Review by yczhangsjtu — APPROVED** [review]

File: src/whir/mod.rs (hunk 1 above)
Comment (chaosma) [reply src/whir/mod.rs:11]: yeah, maybe...

**Pull request merged**